overlord/ifacestate/apparmorprompting/common/common_test.go

```diff
@@ -699,6 +699,18 @@ func (s *commonSuite) TestAbstractPermissionsFromAppArmorFilePermissionsHappy(c
 			notify.AA_MAY_EXEC | notify.AA_MAY_WRITE | notify.AA_MAY_READ,
 			[]string{"read", "write", "execute"},
 		},
+		{
+			// Check that unmapped permissions are discarded without error
```

Comment:
I'm not sure if I am following the new behaviour here. When we say that unmapped and unknown permissions are discarded without error, are we saying that they are auto-denied or auto-allowed? Auto-allow feels problematic from a security point of view and auto-denied feels like it could lead to issues with app behaviour that the user would find difficult to understand / debug.

I'm also not sure I understand what situation leads to us receiving a permission that falls into either of these categories? Surely this is something that we need to have an API contract with the kernel around

Reply:
The new behavior is to exclude the unknown permissions from the prompt. But when the prompt is resolved (either through direct reply or all (known) permissions being satisfied), then all of the originally-requested permissions (including unknown ones) are either allowed or denied accordingly.

So in essence, unmapped permissions are ignored by the prompt, but replies from the listener to the kernel apply to all requested permissions, including unknown ones.

The remaining edge case is if a request contains *only* unknown permissions (or no permissions at all), leaving a prompt with no known permissions. In this case, I think we should auto-deny. And the error message is still logged, as it is when any unknown permission is received.

What do you think of this approach? Samuele is also skeptical, perhaps we should discuss during the next sync. There's no rush to land this now anyway, since #35 has already been merged.

Reply:
Thanks for clarifying!

I think chatting about this at the next sync sounds like a good plan: auto-anything without being transparent to the user about it makes me a little nervous if I'm honest. Even with being transparent this feels like a "we've managed to let something slip through the net" situation rather than a runtime issue :sweat_smile: